apple/Elements/RNSVGImage.mm

```diff
@@ -203,18 +216,16 @@ - (void)setSrc:(RCTImageSource *)src
   _reloadImageCancellationBlock = [[self.bridge moduleForName:@"ImageLoader"]
       loadImageWithURLRequest:src.request
                      callback:^(NSError *error, UIImage *image) {
-                        dispatch_async(dispatch_get_main_queue(), ^{
-                          self->_image = CGImageRetain(image.CGImage);
-                          self->_imageSize = CGSizeMake(CGImageGetWidth(self->_image), CGImageGetHeight(self->_image));
-                          RCTImageSource *sourceLoaded = [src imageSourceWithSize:image.size scale:image.scale];
-                          self->_onLoad(@{
-                              @"width" : @(sourceLoaded.size.width),
-                              @"height" : @(sourceLoaded.size.height),
-                            @"uri" : sourceLoaded.request.URL.absoluteString
-                          });
-                          [self invalidate];
-                        });
-                      }];
+                       dispatch_async(dispatch_get_main_queue(), ^{
+                         self->_image = CGImageRetain(image.CGImage);
+                         self->_imageSize = CGSizeMake(CGImageGetWidth(self->_image), CGImageGetHeight(self->_image));
+                         if (self->_onLoad) {
+                           RCTImageSource *sourceLoaded = [src imageSourceWithSize:image.size scale:image.scale];
+                           self->_onLoad(onLoadParamsForSource(sourceLoaded));
```

Comment:
Does it make sense to add it if we only use it in one place? Or is it potentially going to be reused in other events?

Reply:
without that unfortunately the app crashed in old architecture.

Reply:
I meant `onLoadParamsForSource` method.

Reply:
without that check, the app expects us to pass onLoad prop every time.